core/src/main/java/bisq/core/btc/wallet/BtcWalletService.java

```diff
@@ -548,6 +577,23 @@ public Transaction completePreparedBsqTx(Transaction preparedBsqTx,
         return resultTx;
     }
 
+    private Tuple2<Integer, Integer> getNumInputs(Transaction tx) {
+        int numLegacyInputs = 0;
+        int numSegwitInputs = 0;
+        for (TransactionInput input : tx.getInputs()) {
+            TransactionOutput connectedOutput = input.getConnectedOutput();
+            if (connectedOutput == null || ScriptPattern.isP2PKH(connectedOutput.getScriptPubKey()) ||
```

Comment:
I assume the motivation for treating the connectedOutput == null as legacyInput is that if we dont know the inputs we use the potentially higher fee estimation. If correct, maybe a comment could help to make it more clear.